packages/hoppscotch-app/helpers/import-export/import/openapi.ts

```diff
@@ -177,6 +181,190 @@ const parseOpenAPIV3BodyFormData = (
+    O.getOrElseW(() => "")
+  )
+}
+
+const generateExampleArrayFromOpenAPIV2ItemsObject = (
+  items: OpenAPIV2.ItemsObject
+): RequestBodyExampleType => {
+  // ItemsObject can not hold type "object"
+  // https://swagger.io/specification/v2/#itemsObject
+
+  // TODO : Handle array of objects
+  // https://stackoverflow.com/questions/60490974/how-to-define-an-array-of-objects-in-openapi-2-0
+
+  const primitivePlaceholder = pipe(
+    items,
+    O.fromPredicate(
+      flow((items) => items.type as SchemaType, isSchemaTypePrimitive)
+    ),
+    O.map(getSampleEnumValueOrPlaceholder)
+  )
+
+  if (O.isSome(primitivePlaceholder))
+    return Array.of(primitivePlaceholder.value, primitivePlaceholder.value)
+
+  // If the type is not primitive, it is "array"
+  // items property is required if type is array
+  return Array.of(
+    generateExampleArrayFromOpenAPIV2ItemsObject(
+      items.items as OpenAPIV2.ItemsObject
+    )
+  )
+}
+
+const generateRequestBodyExampleFromOpenAPIV2BodySchema = (
+  schema: OpenAPIV2.SchemaObject
+): RequestBodyExampleType => {
+  if (schema.example) return schema.example as RequestBodyExampleType
+
+  const primitiveTypeExample = pipe(
+    schema,
+    O.fromPredicate(
+      flow(
+        getSchemaTypeFromSchemaObject,
+        O.map(isSchemaTypePrimitive),
+        O.getOrElseW(() => false) // No schema type found in the schema object, assume non-primitive
+      )
+    ),
+    O.map(getSampleEnumValueOrPlaceholder) // Use enum or placeholder to populate primitive field
+  )
+
+  if (O.isSome(primitiveTypeExample)) return primitiveTypeExample.value
+
+  const arrayTypeExample = pipe(
+    schema,
+    O.fromPredicate(
+      flow(
+        getSchemaTypeFromSchemaObject,
+        O.map(isSchemaTypeArray),
+        O.getOrElseW(() => false) // No schema type found in the schema object, assume type to be different from array
+      )
+    ),
+    O.map((schema) => schema.items as OpenAPIV2.ItemsObject),
+    O.map(generateExampleArrayFromOpenAPIV2ItemsObject)
+  )
+
+  if (O.isSome(arrayTypeExample)) return arrayTypeExample.value
+
+  return pipe(
+    schema,
+    O.fromPredicate(
+      flow(
+        getSchemaTypeFromSchemaObject,
+        O.map(isSchemaTypeObject),
+        O.getOrElseW(() => false)
+      )
+    ),
+    O.chain((schema) =>
+      pipe(
+        schema.properties,
+        O.fromNullable,
+        O.map(
+          (properties) =>
+            Object.entries(properties) as [string, OpenAPIV2.SchemaObject][]
+        )
+      )
+    ),
+    O.getOrElseW(() => [] as [string, OpenAPIV2.SchemaObject][]),
+    A.reduce(
+      {} as { [name: string]: RequestBodyExampleType },
+      (aggregatedExample, property) => {
+        const example = generateRequestBodyExampleFromOpenAPIV2BodySchema(
+          property[1]
+        )
+        aggregatedExample[property[0]] = example
+        return aggregatedExample
+      }
+    )
+  )
+}
+
+const getSchemaFromOpenAPIV2Parameter = (
```

Comment:
If this transformation is used a lot then fine, else, I think this function is a bit excessive for a simple usecase.